### ISSUE TITLE
System.Collections.Specialized.Tests.GetItemObjTests.Test01 [FAIL]

### ISSUE THREAD

**Issue by krwq** [dotnet/corefx#1360]
```
     System.Collections.Specialized.Tests.GetItemObjTests.Test01 [FAIL]
        Error, returned wrong item for uppercase key
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release_prtest\src\System.Collections.Specialized\tests\HybridDictionary\GetItemObjTests.cs(328,0): at System.Collections.Specialized.Tests.GetItemObjTests.Test01()
```

I coudln't repro locally, probably some threading issue

**Comment by ellismg** [comment]
Is very likely related to the fact that that we do ToUpper() on the keys and values and expect this to be stable while some of the NameValueCollection tests set the current culture to turkish, which cases "I" differently.

This is racy when they run concurrently.

**Comment by bigodera** [comment]
@ellismg If it really is that(I'm trying to reproduce here), how could we fix it(if there's a way)?

**Comment by stephentoub** [comment]
I expect this was fixed by #4383.  Closing it for now; we can reopen it if it repros again.
